### PR TITLE
add note to exception if 'use_only_tar_bz2' is enabled

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Comes from https://github.com/conda/conda/issues/12332, and https://github.com/conda-incubator/setup-miniconda/issues/267

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/exceptions.py [hunk 1]
```diff
@@ -617,6 +617,13 @@ def __init__(self, packages, channel_urls=()):
 
             and use the search bar at the top of the page.
             """)
+            from .base.context import context
+
```
Comment (dholth) [inline conda/exceptions.py:621]: Let's add a test to this, could go right inside 

```def test_use_only_tar_bz2(platform=OVERRIDE_PLATFORM):
    channel = Channel(join(CHANNEL_DIR, platform))
    SubdirData.clear_cached_local_channel_data()
    with env_var("CONDA_USE_ONLY_TAR_BZ2", True, stack_callback=conda_tests_ctxt_mgmt_def_pol):
        sd = SubdirData(channel)
        precs = tuple(sd.query("zlib"))
        assert precs[0].fn.endswith(".tar.bz2")
```

Only need to raise the exception, and check the message to get code coverage. No need to really "not find" a package.

Comment (jezdez) [reply conda/exceptions.py:621]: I think this hasn't reached the level of documentation, but it'd be worth it

File: conda/exceptions.py [hunk 2]
```diff
@@ -617,6 +617,13 @@ def __init__(self, packages, channel_urls=()):
 
             and use the search bar at the top of the page.
             """)
+            from .base.context import context
```
Comment (jezdez) [inline conda/exceptions.py:620]: Do we need to import this inline? This is a little unfortunate.

Comment (jaimergp) [reply conda/exceptions.py:620]: See precedent in other exceptions. I am not sure why, maybe import times?

File: conda/exceptions.py (hunk 1 above)
Comment (jaimergp) [reply conda/exceptions.py:621]: Added test in `test_exceptions.py`.

File: tests/test_exceptions.py [hunk 3]
```diff
@@ -490,3 +492,17 @@ def test_BinaryPrefixReplacementError(self):
           original data Length: 1404
           new data length: 1104
         """).strip()
+
+    @patch("conda.exceptions.os.isatty", return_value=True)
+    def test_PackagesNotFoundError_use_only_tar_bz2(self, isatty_mock):
+        note = "use_only_tar_bz2"
+        for use_only_tar_bz2 in (True, False):
+            expected = ["", note][use_only_tar_bz2]
```
Comment (dholth) [inline tests/test_exceptions.py:500]: @jaimergp we could do it this way. We don't have the requirement to "not show the note" when the flag is not set, and this regex matcher is bad at negative testing.

Comment (jaimergp) [reply tests/test_exceptions.py:500]: As you wish, but I find this way of assigning booleans a bit unreadable. I'd rather see this in code I am maintaining:

```suggestion
            expected = "use_only_tar_bz2" if use_only_tar_bz2 else ""
```

The `raises` + `match` combo is killer though.

File: tests/test_exceptions.py [hunk 4]
```diff
@@ -490,3 +492,17 @@ def test_BinaryPrefixReplacementError(self):
           original data Length: 1404
           new data length: 1104
         """).strip()
+
+    @patch("conda.exceptions.os.isatty", return_value=True)
```
Comment (dholth) [inline tests/test_exceptions.py:496]: this patch is probably useless now

Comment (jaimergp) [reply tests/test_exceptions.py:496]:
```suggestion
```

**Review by jezdez — APPROVED** [review]

**Review by dholth — DISMISSED** [review]
muy bien

**Comment by dholth** [comment]
Since we are using squash merge, and because I'm bad at rebase, I've been using merge-from-main and never rebase-and-force-push

**Review by dholth — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Review by dholth — APPROVED** [review]

**Pull request merged**